src/main/java/com/GG/T9AgeCombat/predicates/CheckLimitationPredicate.java

```diff
@@ -1,19 +1,60 @@
 package com.GG.T9AgeCombat.predicates;
 
+import com.GG.T9AgeCombat.enums.EquipmentCategoryEnum;
+import com.GG.T9AgeCombat.enums.EquipmentTypeEnum;
+import com.GG.T9AgeCombat.enums.SpecialRulePropertyEnum;
+import com.GG.T9AgeCombat.enums.UnitTypeEnum;
+import com.GG.T9AgeCombat.models.Equipment;
+import com.GG.T9AgeCombat.models.SpecialRuleProperty;
 import org.springframework.stereotype.Component;
 
+import java.util.List;
+
 import static com.GG.T9AgeCombat.common.Constants.HORDE_MODELS_PER_RANK;
+import static java.util.stream.Collectors.toList;
 
 @Component
 public class CheckLimitationPredicate {
-    private CheckLimitationPredicate() {
+
+    public CheckLimitationPredicate() {
     }
 
     public static boolean checkFirstRound(Boolean isFirstRound) {
         return isFirstRound;
     }
 
+    public static boolean checkFirstRoundNotCharging(Boolean isFirstRound, boolean isCharging) {
+        return isFirstRound && !isCharging;
+    }
+
+    public static boolean passDisciplineTest(int leadership, Integer roll) {
+        return roll <= leadership;
+    }
+
     public static boolean checkHorde(Integer unitWidth) {
         return unitWidth >= HORDE_MODELS_PER_RANK;
     }
+
+    public static boolean checkLightningReflexes(List<SpecialRuleProperty> unitSpecialRules) {
```

Comment:
Since we always negate the return from this method I think it would make sense to invert the method rather than adding a `!` everywhere we call it (i.e. change the name and return the negated result).

Same thing with the great weapon.